avocado/utils/asset.py

```diff
@@ -367,7 +367,7 @@ def fetch(self):
                                       self.relative_dir)
             dirname = os.path.dirname(asset_file)
             if not os.path.isdir(dirname):
-                os.makedirs(dirname)
+                os.makedirs(dirname, exist_ok=True)
```

Comment:
Hi @beraldoleal, IIUC when the file already exists, we don't have to downloaded again, because it is downloaded by different thread, and we can return it.

Reply:
Hi @richtja, Look at the error on #4631. For some reason this line was hit by multiple threads. Adding this `exist_ok` fixed the issue.  `self._path` was not set properly, so the second `.get()` is called while the first one is still downloading.

Reply:
My point is that the problem is in two threads downloading the same image, and we should avoid such behavior. As I see it, your change removes the error, but the problem will remain. But I didn't test it, so maybe I get the problem wrong.

Reply:
I understand your point, and this is not fixing the real problem, but instead is just avoiding a crash.

As a test writer I should not get concerned about downloading two images at the same time. For instance I can write two tests that needs the same Fedora 34 image, but I'm relying on avocado to create the snapshots for me.

But I agree with you, we should work on that properly, but this could be left to another PR. This PR is just removing the crash.

Reply:
If the problem is simultaneous downloads of the same image, we need an issue to have that tracked and fixed. This will fix the crash, but as @richtja said, this is not solving the problem.